projects/storefrontlib/src/cms-components/checkout/components/checkout-progress/checkout-progress-mobile-bottom/checkout-progress-mobile-bottom.component.spec.ts

```diff
@@ -1,41 +1,74 @@
-import { Pipe, PipeTransform } from '@angular/core';
+import { Pipe, PipeTransform, Type } from '@angular/core';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
 import { RouterTestingModule } from '@angular/router/testing';
-import { StoreModule } from '@ngrx/store';
 import {
   I18nTestingModule,
-  RoutesConfig,
   RoutingConfigService,
   RoutingService,
 } from '@spartacus/core';
 import { Observable, of } from 'rxjs';
-import { defaultStorefrontRoutesConfig } from '../../../../../cms-structure/routing/default-routing-config';
-import { CheckoutConfig } from '../../../config/checkout-config';
-import { defaultCheckoutConfig } from '../../../config/default-checkout-config';
+import { CheckoutStep, CheckoutStepType } from '../../../model';
+import { CheckoutConfigService } from '../../../services/checkout-config.service';
+import { BehaviorSubject } from 'rxjs';
 import { CheckoutProgressMobileBottomComponent } from './checkout-progress-mobile-bottom.component';
 
-const MockCheckoutConfig: CheckoutConfig = defaultCheckoutConfig;
-const MockRoutesConfig: RoutesConfig = defaultStorefrontRoutesConfig;
+const mockCheckoutSteps: Array<CheckoutStep> = [
+  {
+    id: 'step0',
+    name: 'step 0',
+    routeName: 'route0',
+    type: [CheckoutStepType.PAYMENT_TYPES],
+    enabled: true,
+  },
+  {
+    id: 'step1',
+    name: 'step 1',
+    routeName: 'route1',
+    type: [CheckoutStepType.SHIPPING_ADDRESS],
+    enabled: true,
+  },
+  {
+    id: 'step2',
+    name: 'step 2',
+    routeName: 'route2',
+    type: [CheckoutStepType.DELIVERY_MODE],
+    enabled: true,
+  },
+];
+
+class MockCheckoutConfigService {
+  steps$: BehaviorSubject<CheckoutStep[]> = new BehaviorSubject<CheckoutStep[]>(
+    mockCheckoutSteps
+  );
+}
 
+class MockRoutingConfigService {
+  getRouteConfig(stepRoute) {
+    if (stepRoute === 'route0') {
```

Comment:
Maybe instead of the ifs you can check for not null and just return "checkout/${stepRoute}" ? Just a style change

Reply:
I just use some mock data like "route0" and "checkout/route0", the `stepRoute` and `paths` do not have any relationship.